addons/missileguidance/functions/fnc_seekerType_MCLOS.sqf

```diff
@@ -16,16 +16,39 @@
  * Public: No
  */
 params ["", "_args"];
-_args params ["_firedEH", "", "_flightParams"];
-_firedEH params ["","","","","","","_projectile","_gunner"];
+_args params ["_firedEH", "", "_flightParams", "", "_stateParams"];
+_firedEH params ["_shooter","","","","","","_projectile","_gunner"];
 _flightParams params ["_pitchRate", "_yawRate"];
+_stateParams params ["", "_seekerStateParams"];
 
 private _projectilePos = getPosASLVisual _projectile;
 private _source = _projectile getVariable [QGVAR(source), _projectile];
+private _mclosInput = (_source getVariable [QGVAR(MCLOS_direction), [0, 0, 0]]);
 
-private _accelerationDirection = _projectile vectorModelToWorldVisual (_source getVariable [QGVAR(MCLOS_direction), [0, 0, 0]]);
+if (_seekerStateParams isNotEqualTo []) then { // Handle AI
+    _seekerStateParams params ["_launchPos", "_launchVec"];
+    if (!(alive _shooter && {alive _gunner} && {(vehicle _gunner) == _shooter} && {_gunner call EFUNC(common,isAwake)})) exitWith {};
+
+    private _distance = _launchPos vectorDistance _projectilePos;
+    private _projectedPath = _launchPos vectorAdd (_launchVec vectorMultiply _distance);
```

Comment:
this only aims for the point the ai was aiming when they fired. if the target moves, this inherently misses. i think we could do something similar to SACLOS where we figure out the position the gunner is looking but as SACLOS shows that isn't easy. this is good for now, just wanted to note this